modules/contextualization/cdf_file_annotation/functions/fn_file_annotation_finalize/handler.py

```diff
@@ -0,0 +1,184 @@
+import sys
+import threading
+from datetime import datetime, timezone, timedelta
+from cognite.client import CogniteClient
+
+from dependencies import (
+    create_config_service,
+    create_logger_service,
+    create_write_logger_service,
+    create_general_report_service,
+    create_general_retrieve_service,
+    create_general_apply_service,
+    create_general_pipeline_service,
+)
+from services.FinalizeService import AbstractFinalizeService, GeneralFinalizeService
+from services.ApplyService import IApplyService
+from services.ReportService import IReportService
+from services.RetrieveService import IRetrieveService
+from services.PipelineService import IPipelineService
+from utils.DataStructures import PerformanceTracker
+
+
+def handle(data: dict, client: CogniteClient) -> dict:
+    """
+    Main entry point for the cognite function.
+    1. Create an instance of config, logger, and tracker
+    2. Create an instance of the finalize function and create implementations of the interfaces
+    3. Run the finalize instance until...
+        4. It's been 7 minutes
+        5. There are no jobs left to process
+    6. Generate a report that includes capturing the annotations in RAW
+    NOTE: Cognite functions have a run-time limit of 10 minutes.
+    Don't want the function to die at the 10minute mark since there's no guarantee all code will execute.
+    Thus we set a timelimit of 7 minutes (conservative) so that code execution is guaranteed.
+    """
+    start_time = datetime.now(timezone.utc)
+    log_level = data.get("logLevel", "INFO")
+
+    config_instance, client = create_config_service(function_data=data, client=client)
+    logger_instance = create_logger_service(log_level)

```

Comment:
you should just call this logger instead of logger_instance. Just because there is a need to create your own doesn't mean you can't follow the rest of the pattern set up by the built in logger.
https://docs.python.org/3/howto/logging.html